pages/components/Footer/Footer.js

```diff
@@ -1,58 +1,70 @@
-import styles from "./Footer.module.scss"
+import S from "./Footer.module.scss"
 import Link from "next/link"
-import { useState } from "react" // Импортируем useState
-
-//to do:
-// Add popup "not validation email"
+import {useState} from "react"
+import EmailValidationMessage from "@/pages/components/EmailValidationMessage"
+import SubscriptionSuccessMessage from "@/pages/components/SubscriptionSuccessMessage"
+import {EMAIL_REGEXP} from "@/pages/utils/varibles"
 
 const Footer = () => {
-  const [ email, setEmail ] = useState("") // Создаем состояние для хранения значения поля email
-  const [ emailStatus, setEmailStatus ] = useState(false)
+  const [email, setEmail] = useState("")
+  const [emailStatus, setEmailStatus] = useState(false)
+  const [validation, setValidation] = useState(false)
+
+  const submitEmail = () => {
+    setEmailStatus(true)
+  }
+
+  const showEmailValidationError = () => {
+    setValidation(true)
+  }
+
   const handleClick = (e) => {
     e.preventDefault()
-
-    if ( isValidEmail(email) ) {
-      setEmailStatus(true)
-      console.log("Email отправлен:", email)
-    } else {
-      alert("Пожалуйста, введите действительный E-mail.")
-    }
+    isValidEmail(email) ? submitEmail() : showEmailValidationError()
   }
 
   const isValidEmail = (value) => {
-    const emailPattern = /^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,}$/i
-    return emailPattern.test(value)
+    EMAIL_REGEXP.test(value) ? setValidation(false) : setValidation(true)
+    return EMAIL_REGEXP.test(value)
   }
 
+
   return (
     <footer>
-      <h3 className={ styles.heading }>ПОДПИСЫВАЙСЯ НА НАШИ НОВОСТИ</h3>
+      <h3 className={S.heading}>ПОДПИСЫВАЙСЯ НА НАШИ НОВОСТИ</h3>
 
-      { !emailStatus ? (
-        <form className={ styles.follow }>
+      {!emailStatus ? <>
+        <form className={S.follow}>
           <input
             pattern="^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,}$"
```

Comment:
move constant to the top of the file

`const EMAIL_PATTERN = "^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,}$"`